sp/src/game/shared/mapbase/matchers.cpp

```diff
@@ -5,45 +5,17 @@
 // $NoKeywords: $
 //=============================================================================
 
+// Must be included before cbase.h, otherwise some macros break the Linux build.
+#include <regex>
+
 #include "cbase.h"
 
 #include "matchers.h"
-#include "fmtstr.h"
-#include <regex>
+//#include "fmtstr.h"
 
 ConVar mapbase_wildcards_enabled("mapbase_wildcards_enabled", "1", FCVAR_NONE, "Toggles Mapbase's '?' wildcard and true '*' features. Useful for maps that have '?' in their targetnames.");
 ConVar mapbase_regex_enabled("mapbase_regex_enabled", "1", FCVAR_NONE, "Toggles Mapbase's regex matching handover.");
 
-#ifdef CLIENT_DLL
-// FIXME: There is no clientside equivalent to the RS code
-bool ResponseSystemCompare(const char *criterion, const char *value) { return Matcher_NamesMatch(criterion, value); }
-#else
-extern bool ResponseSystemCompare(const char *criterion, const char *value);
-#endif
-
-//=============================================================================
-// These are the "matchers" that compare with wildcards ("any*" for text starting with "any")
-// and operators (<3 for numbers less than 3).
-// 
-// Matcher_Match - Matching function using RS operators and NamesMatch wildcards/regex.
-// Matcher_Regex - Uses regex functions from the std library.
-// Matcher_NamesMatch - Based on Valve's original NamesMatch function, using wildcards and regex.
-// 
-// AppearsToBeANumber - Response System-based function which checks if the string might be a number.
-//=============================================================================
-
-inline bool Matcher_Match(const char *pszQuery, const char *szValue)
```

Comment:
These functions were inlined because the original `NamesMatch()` (the Valve function these matchers are based on) was `FORCEINLINE`. I was planning on removing the `inline`/``FORCEILINE` on most of these functions and cleaning up the header file at some point, but I didn't realize it was already causing problems on Linux.

I would prefer removing the inlining on most of the larger functions and moving them back to the CPP file.

Reply:
I see. I was doubting they're good candidates for inlining so I concur that moving them back is good.